app/controllers/posts_controller.rb

```diff
@@ -20,9 +20,7 @@ def create
   private
 
   def timeline_posts
-    result = Post.where(user: current_user.friends)
-    result += current_user.posts
-    @timeline_posts ||= result.sort { |x, y| y.created_at <=> x.created_at }
+    @timeline_posts ||= Post.all.ordered_by_most_recent.includes(:user)
   end
```

Comment:
- [ ] Post should include both user and friends 

````ruby
def friends_and_own_posts
  Post.where(user: (self.friends + self))
  # This will produce SQL query with IN. Something like: select * from posts where user_id IN (1,45,874,43);
end

````